qiskit/backends/providerutils.py

```diff
@@ -41,6 +41,14 @@ def _match_all(dict_, criteria):
     status_filters = {}
     for key, value in kwargs.items():
         if all(key in backend.configuration() for backend in backends):
+            if key == 'name':
```

Comment:
Can you revert this change, and introduce back the 
```
# Special handling of the `name` parameter, to support alias resolution.
```
block in `ibmqprovider.py`? This `_match_all()` function should be general and not need knowledge of the ibmq internals, and will probably end up causing circular import problems plus breaking the intended modularization.

Reply:
Indeed, that removal was done before I realized that moving to depreciated backnames is the thing to do.

Reply:
Cool, thanks for the speedy update!